projects/templates/src/lib/components/po-page-customization/po-page-customization.service.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { TestBed, async, fakeAsync, tick } from '@angular/core/testing';
+
+import { PoPageCustomizationService } from './po-page-customization.service';
+import { HttpTestingController, HttpClientTestingModule } from '@angular/common/http/testing';
+import { PoPageOptions } from './po-page-options.interface';
+
+const originalPageOptions: PoPageOptions = {
+  title: 'Original Title',
+  breadcrumb: {
+    items: [
+      { label: 'Home' },
+      { label: 'Hiring processes' }
+    ]
+  },
+  actions: [
+    { label: 'Feature 1', url: '/feature1' },
+    { label: 'Feature 2', url: '/feature2' }
+  ],
+  filters: [
+    { property: 'filter1' },
+    { property: 'filter2' }
+  ]
+};
+
+const mergedPageOptions: PoPageOptions = {
+  title: 'New Title',
+  breadcrumb: {
+    items: [
+      { label: 'Test' },
+      { label: 'Test2' }
+    ]
+  },
+  actions: [
+    { label: 'Feature 1', url: '/new-feature1' },
+    { label: 'Feature 2', url: '/feature2' },
+    { label: 'Feature 3', url: '/new-feature3' }
+  ],
+  filters: [
+    { property: 'filter1' },
+    { property: 'filter2' },
+    { property: 'filter3' }
+  ]
+};
+
+const newPageOptions: PoPageOptions = {
+  title: 'New Title',
+  breadcrumb: {
+    items: [
+      { label: 'Test' },
+      { label: 'Test2' }
+    ]
+  },
+  actions: [
+    { label: 'Feature 1', url: '/new-feature1' },
+    { label: 'Feature 3', url: '/new-feature3' }
+  ],
+  filters: [
+    { property: 'filter1' },
+    { property: 'filter3' }
+  ]
+};
+
+function customOptionFunctionMock(): PoPageOptions {
+  return newPageOptions;
+}
+
+describe('PoPageCustomizationService:', () => {
+  let poPageCustomizationService: PoPageCustomizationService;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [PoPageCustomizationService]
+    });
+
+    poPageCustomizationService = TestBed.get(PoPageCustomizationService);
+    httpMock = TestBed.get(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should be created', () => {
+    expect(poPageCustomizationService instanceof PoPageCustomizationService).toBeTruthy();
+  });
+
+  describe('Methods: ', () => {
+    describe('changeOriginalOptionsToNewOptions:', () => {
```

Comment:
Temos como uma convenção, criar os describe para testes apenas se tiver configurações e alguns testes (it) referente a ele.

Reply:
Feito